### PR TITLE
Add possibility to search Blog::Post and Page at the same time + i18n issue

### PR DESCRIPTION
**Pull request by keklesurvivant**
Hi,

I would like to use elasticsearch to search both Blog::Post and Page. But when I faced issues when I attempted to generate the index, but I was able to find the root cause quickly.

I'm not sure about the fix related to the rake task. Do not hesitate if you have any remark.

Kévin

### CODE REVIEW

File: lib/tasks/refinery/elasticsearch.rake [hunk 1]
```diff
@@ -15,6 +15,9 @@ namespace :refinery do
 
     desc 'Recreate Elasticsearch index'
     task recreate: :setup_logger do
+      # Use the right language
+      ::I18n.locale = ::I18n.default_locale
```
Comment (bricesanchez) [inline lib/tasks/refinery/elasticsearch.rake:19]: What happens if i have multiple locales ? Should not we loop each locale ? What do you think @datenimperator ?

File: app/decorators/models/refinery/page_decorator.rb [hunk 2]
```diff
@@ -34,7 +34,7 @@ def to_index
       end
 
       def self.indexable
-        live.in_menu
+        live
```
Comment (bricesanchez) [inline app/decorators/models/refinery/page_decorator.rb:39]: I'm ok with this change.

File: app/decorators/models/refinery/page_decorator.rb [hunk 3]
```diff
@@ -6,7 +6,7 @@
       define_mapping do
         {
           title: { type: 'string' },
-          browser_title: { type: 'string' },
+          browser_title: { type: 'string', analyzer: 'snowball' },
```
Comment (bricesanchez) [inline app/decorators/models/refinery/page_decorator.rb:9]: I'm ok with this change.

File: lib/tasks/refinery/elasticsearch.rake (hunk 1 above)
Comment (datenimperator) [reply lib/tasks/refinery/elasticsearch.rake:19]: Indexing multiple locales is surprisingly hard. Basically you'd have to index multiple versions of the same document. Not supported now, but might be in the future.

Comment (bricesanchez) [reply lib/tasks/refinery/elasticsearch.rake:19]: For my multilang project i use this (ugly) config : 

``` ruby
# app/decorators/models/refinery/page_decorator.rb

Refinery::Page.class_eval do
  include ::Refinery::Elasticsearch::Searchable

  define_mapping do
    {
      title_fr: { type:'string', boost:2.0 },
      title_en: { type:'string', boost:2.0 },
      browser_title: { type:'string' },
      menu_title_fr: { type:'string' },
      menu_title_en: { type:'string' },
      meta_description: { type:'string' },
      part: {
        type:'object',
        properties: {
          body_fr: { type:'string', analyzer:'snowball' },
          body_en: { type:'string', analyzer:'snowball' }
        }
      },
      created_at: { type:'date' },
      updated_at: { type:'date' }
    }
  end

  def to_index
    if in_menu?
      {
        title_fr: self.title_fr,
        title_en: self.title_en,
        part: self.parts.map{ |part|
          {
            body_fr: strip_tags(part.body_fr),
            body_en: strip_tags(part.body_en)
          }
        },
        browser_title: self.browser_title,
        menu_title_fr: self.menu_title_fr,
        menu_title_en: self.menu_title_en,
        meta_description: self.meta_description,
        created_at: self.created_at,
        updated_at: self.updated_at
      }
    end
  end

  def self.indexable
    live.in_menu
  end
end
```

i've no changes in the rake task.

**Pull request closed without merging**